tests/system_tests_multicast.py

```diff
@@ -429,6 +449,245 @@ def test_999_check_for_leaks(self):
+               [('connector', {'name': 'connectorToA',
+                               'role': 'inter-router',
+                               'port': interrouter_port_B_to_A}),
+                ('listener', {'role': 'inter-router',
+                              'port': interrouter_port_C_to_B}),
+                ('listener', {'role': 'edge',
+                              'port': cls.INTB_edge_port})])
+        cls.INT_B = cls.routers[1]
+        cls.INT_B.listener = cls.INT_B.addresses[0]
+
+        router('EA1', 'edge',
+               [('listener', {'name': 'rc', 'role': 'route-container',
+                              'port': cls.tester.get_port()}),
+                ('connector', {'name': 'uplink', 'role': 'edge',
+                               'port': cls.INTA_edge_port})])
+        cls.EA1 = cls.routers[2]
+        cls.EA1.listener = cls.EA1.addresses[0]
+        cls.EA1.route_container = cls.EA1.addresses[1]
+
+        router('EB1', 'edge',
+               [('connector', {'name': 'uplink',
+                               'role': 'edge',
+                               'port': cls.INTB_edge_port,
+                               'maxFrameSize': 1024}),
+                ('listener', {'name': 'rc', 'role': 'route-container',
+                              'port': cls.tester.get_port()})])
+        cls.EB1 = cls.routers[3]
+        cls.EB1.listener = cls.EB1.addresses[0]
+        cls.EB1.route_container = cls.EB1.addresses[1]
+
+        router('INT.C', 'interior',
+               [('connector', {'name': 'connectorToA',
+                               'role': 'inter-router',
+                               'port': interrouter_port_C_to_A}),
+                ('connector', {'name': 'connectorToB',
+                               'role': 'inter-router',
+                               'port': interrouter_port_C_to_B}),
+                ('listener', {'role': 'edge',
+                              'port': cls.INTC_edge_port})])
+        cls.INT_C = cls.routers[4]
+        cls.INT_C.listener = cls.INT_C.addresses[0]
+
+        router('EC1', 'edge',
+               [('connector', {'name': 'uplink',
+                               'role': 'edge',
+                               'port': cls.INTC_edge_port,
+                               'maxFrameSize': 1024}),
+                ('listener', {'name': 'rc', 'role': 'route-container',
+                              'port': cls.tester.get_port()})])
+        cls.EC1 = cls.routers[5]
+        cls.EC1.listener = cls.EC1.addresses[0]
+        cls.EC1.route_container = cls.EC1.addresses[1]
+
+        cls.INT_A.wait_router_connected('INT.B')
+        cls.INT_A.wait_router_connected('INT.C')
+        cls.INT_B.wait_router_connected('INT.A')
+        cls.INT_B.wait_router_connected('INT.C')
+        cls.INT_C.wait_router_connected('INT.A')
+        cls.INT_C.wait_router_connected('INT.B')
+        cls.EA1.wait_connectors()
+        cls.EB1.wait_connectors()
+        cls.EC1.wait_connectors()
```

Comment:
@Karen-Schoener you might also want to use the `is_edge_routers_connected` on each interior router as seen here - 

https://github.com/skupperproject/skupper-router/blob/main/tests/system_tests_tcp_adaptor.py#L609

Reply:
Thanks for mentioning.

The new test seems to run more reliably with this additional check.

In an earlier rev of the test (without this check), sometimes 1 of the receivers reported 0 multicast message received...

Reply:
> The new test seems to run more reliably with this additional check.

Yay !!